RecoLocalTracker/SiPixelClusterizer/plugins/gpuClustering.h

```diff
@@ -30,137 +30,155 @@ namespace gpuClustering {
     }
   }
 
-  __global__ void findClus(uint16_t const * id,
-                           uint16_t const * x,
-                           uint16_t const * y,
-                           uint16_t const * adc,
-                           uint32_t const * moduleStart,
-                           uint32_t * clusInModule, uint32_t * moduleId,
-                           int32_t * clus,
+  __global__ void findClus(uint16_t const * id,             // module id of each pixel
+                           uint16_t const * x,              // local coordinates of each pixel
+                           uint16_t const * y,              //
+                           uint32_t const * moduleStart,    // index of the first pixel of each module
+                           uint32_t * nClustersInModule,    // output: number of clusters found in each module
+                           uint32_t * moduleId,             // output: module id of each module
+                           int32_t *  clusterId,            // output: cluster id of each pixel
                            int numElements)
   {
-    __shared__ bool go;
-    __shared__ int nclus;
-    __shared__ int msize;
-
     if (blockIdx.x >= moduleStart[0])
       return;
 
-    auto first = moduleStart[1 + blockIdx.x];
-    auto me = id[first];
-
-    assert(me < MaxNumModules);
+    auto firstPixel = moduleStart[1 + blockIdx.x];
+    auto thisModuleId = id[firstPixel];
+    assert(thisModuleId < MaxNumModules);
 
 #ifdef GPU_DEBUG
-    if (me%100 == 1)
+    if (thisModuleId % 100 == 1)
       if (threadIdx.x == 0)
-        printf("start clusterizer for module %d in block %d\n", me, blockIdx.x);
+        printf("start clusterizer for module %d in block %d\n", thisModuleId, blockIdx.x);
 #endif
 
-    first += threadIdx.x;
-    if (first>= numElements)
-      return;
```

Comment:
For my education, why is this replaced with the
```cpp
bool active = (first < numElements);
if (active) {
```
?

Reply:
I am unsure how `__syncthreads` is affected by threads that have `return`ed, so I have removed all `return` statements.
I can see if putting them back reintroduces a problem or not...

Reply:
Ok, thanks for the explanation!